common/src/main/java/gov/cms/ab2d/common/service/JobServiceImpl.java

```diff
@@ -56,7 +69,37 @@ public Job getJobByJobID(String jobID) {
         return job;
     }
 
+    @Override
     public Job updateJob(Job job) {
         return jobRepository.save(job);
     }
+
+    @Override
+    public Resource getResourceForJob(String jobID, String fileName) throws MalformedURLException {
+        Job job = getJobByJobID(jobID);
+
+        // Make sure that there is a path that matches a job output for the job they are requesting
+        boolean jobOutputMatchesPath = false;
+        for (JobOutput jobOutput : job.getJobOutput()) {
+            if (jobOutput.getFilePath().equals(fileName)) {
+                jobOutputMatchesPath = true;
+                break;
+            }
+        }
+
+        if (!jobOutputMatchesPath) {
+            throw new ResourceNotFoundException("No Job Output with the file name " + fileName + " exists in our records");
+        }
+
+        Path file = Paths.get(fileDownloadPath + fileName);
```

Comment:
Has to be `fileDownloadPath+ jobId + FileUtils.pathSeparator + fileName`. This is because each job will get its own sub-directory under the EFS root.